iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkImpl.java

```diff
@@ -636,162 +633,39 @@ public BusViewImpl getBusView() {
-            // trigger a new component computation and so on a change in the value of the num component already passed
-            // (and outdated consequently) in parameter of this method
-            return num != -1 ? components.get(num) : null;
-        }
-
-    }
-
-    static final class ConnectedComponentsManager extends AbstractComponentsManager<ConnectedComponentImpl> {
-
-        private ConnectedComponentsManager(NetworkImpl network) {
+        private ConnectedComponentsManager(Network network) {
             super(network);
         }
 
         @Override
-        protected void fillAdjacencyList(Map<String, Integer> id2num, TIntArrayList[] adjacencyList) {
-            super.fillAdjacencyList(id2num, adjacencyList);
-            for (HvdcLineImpl line : network.index.getAll(HvdcLineImpl.class)) {
-                BusExt bus1 = line.getConverterStation1().getTerminal().getBusView().getBus();
-                BusExt bus2 = line.getConverterStation2().getTerminal().getBusView().getBus();
-                addToAdjacencyList(bus1, bus2, id2num, adjacencyList);
-            }
-        }
-
-        @Override
-        protected String getComponentLabel() {
-            return "Connected";
+        protected ConnectedComponentImpl createComponent(int num, int size) {
+            return new ConnectedComponentImpl(num, size, ((NetworkImpl) network).ref);
         }
 
         @Override
-        protected void setComponentNumber(BusExt bus, int num) {
+        protected void setComponentNumber(Bus bus, int num) {
             Objects.requireNonNull(bus);
-            bus.setConnectedComponentNumber(num);
-        }
-
-        protected ConnectedComponentImpl createComponent(int num, int size) {
-            return new ConnectedComponentImpl(num, size, network.ref);
+            ((BusExt) bus).setConnectedComponentNumber(num);
         }
     }
 
-    static final class SynchronousComponentsManager extends AbstractComponentsManager<ComponentImpl> {
+    static final class SynchronousComponentsManager extends AbstractSynchronousComponentsManager<ComponentImpl> {
 
-        private SynchronousComponentsManager(NetworkImpl network) {
+        private SynchronousComponentsManager(Network network) {
             super(network);
         }
 
-        protected ComponentImpl createComponent(int num, int size) {
-            return new ComponentImpl(num, size, network.ref);
-        }
-
         @Override
-        protected String getComponentLabel() {
-            return "Synchronous";
+        protected ComponentImpl createComponent(int num, int size) {
+            return new ComponentImpl(num, size, ((NetworkImpl) network).ref);
         }
 
         @Override
-        protected void setComponentNumber(BusExt bus, int num) {
+        protected void setComponentNumber(Bus bus, int num) {
             Objects.requireNonNull(bus);
-            bus.setSynchronousComponentNumber(num);
+            ((BusExt) bus).setSynchronousComponentNumber(num);
```

Comment:
Maybe the cast is not necessary: setConnectedComponentNumber is visible in the public API?